sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingAggregationSuite.scala

```diff
@@ -686,6 +659,57 @@ class StreamingAggregationSuite extends StateStoreMetricsTest with Assertions {
     }
   }
 
+  testQuietlyWithAllStateVersions("changing schema of state when restarting query -" +
+    " schema check off", (SQLConf.STATE_SCHEMA_CHECK_ENABLED.key, "false")) {
+    withTempDir { tempDir =>
+      val (inputData, aggregated) = prepareTestForChangingSchemaOfState(tempDir)
+
+      testStream(aggregated, Update())(
+        StartStream(checkpointLocation = tempDir.getAbsolutePath),
+        AddData(inputData, 21),
+        ExpectFailure[SparkException] { e =>
+          val cause = e.getCause
+          // it would bring other error in runtime, but it shouldn't check schema in any way
+          assert(!cause.isInstanceOf[StateSchemaNotCompatible])
```

Comment:
Note that it anyway throws exception even disabling schema check, and the exception would be non-friendly one. That's what this patch brings the value.